internal/conf/computed.go

```diff
@@ -294,6 +294,57 @@ func AuthMinPasswordLength() int {
 	return val
 }
 
+// GenericPasswordPolicy a generic password policy that holds password requirements
+type GenericPasswordPolicy struct {
+	Enabled                   bool
+	MinimumLength             int
+	NumberOfSpecialCharacters int
+	RequireAtLeastOneNumber   bool
+	RequireUpperandLowerCase  bool
+}
+
+// AuthPasswordPolicy returns a GenericPasswordPolicy for password validation
+func AuthPasswordPolicy() GenericPasswordPolicy {
+
+	p := Get().AuthPasswordPolicy
+	ml := Get().AuthMinPasswordLength
+
+	if p == nil {
+		ep := ExperimentalFeatures().PasswordPolicy
+
+		if ep != nil {
+			return GenericPasswordPolicy{
+				Enabled:                   ep.Enabled,
+				MinimumLength:             ml,
+				NumberOfSpecialCharacters: ep.NumberOfSpecialCharacters,
+				RequireAtLeastOneNumber:   ep.RequireAtLeastOneNumber,
+				RequireUpperandLowerCase:  ep.RequireUpperandLowerCase,
+			}
+		}
+	} else {
+		return GenericPasswordPolicy{
+			Enabled:                   p.Enabled,
+			MinimumLength:             ml,
+			NumberOfSpecialCharacters: p.NumberOfSpecialCharacters,
+			RequireAtLeastOneNumber:   p.RequireAtLeastOneNumber,
+			RequireUpperandLowerCase:  p.RequireUpperandLowerCase,
+		}
+	}
+
+	return GenericPasswordPolicy{
+		Enabled:                   false,
+		MinimumLength:             0,
+		NumberOfSpecialCharacters: 0,
+		RequireAtLeastOneNumber:   false,
+		RequireUpperandLowerCase:  false,
+	}
```

Comment:
I'd prefer early-returns to make the control flow easier to understand (in general but here especially):

```suggestion
	ml := Get().AuthMinPasswordLength

	if p := Get().AuthPasswordPolicy; p != nil {
		return GenericPasswordPolicy{
			Enabled:                   p.Enabled,
			MinimumLength:             ml,
			NumberOfSpecialCharacters: p.NumberOfSpecialCharacters,
			RequireAtLeastOneNumber:   p.RequireAtLeastOneNumber,
			RequireUpperandLowerCase:  p.RequireUpperandLowerCase,
		}
	}

	if ep := ExperimentalFeatures().PasswordPolicy; ep != nil {
		return GenericPasswordPolicy{
			Enabled:                   ep.Enabled,
			MinimumLength:             ml,
			NumberOfSpecialCharacters: ep.NumberOfSpecialCharacters,
			RequireAtLeastOneNumber:   ep.RequireAtLeastOneNumber,
			RequireUpperandLowerCase:  ep.RequireUpperandLowerCase,
		}
	}

	return GenericPasswordPolicy{
		Enabled:                   false,
		MinimumLength:             0,
		NumberOfSpecialCharacters: 0,
		RequireAtLeastOneNumber:   false,
		RequireUpperandLowerCase:  false,
	}
```